VERSIONING.md

```diff
@@ -0,0 +1,71 @@
+# NUnit Console Runner and Engine Versioning
+
+Both **Console** and **Engine** packages use a version number in the form MAJOR.MINOR.PATCH. We change the
+
+- MAJOR version when we have made incompatible API changes
+- MINOR version when adding functionality in a backwards-compatible manner
+- PATCH version when you making backwards-compatible bug fixes
+
+This versioning approach is, of course, what is known as "Semantic Versioning." Taken by itself, however, the above description leaves out some rather important stuff...
+
+* What exactly are the APIs we are protecting from incompatible changes?
+* What degree of change is needed in each of those APIs for it to be considered a breaking change?
+
+The remainder of this document attempts to answer those questions.
+
+## Console Runner API
+
+First, let's note that the **Console Runner** is not designed to be used as a library, so there is no ABI or library-like API to worry about. However, the runner does present a command-line interface, which is used both interactively and through user-created scripts. The command-line API is defined by our documentation, including the help message the runner displays and the [published documentation](https://docs.nunit.org/articles/nunit/running-tests/Console-Command-Line.html) on our website.
+
+### Examples of MAJOR Command-Line Changes
+
+- Complete removal of an option.
+- Changing the behavior of an option so that it no longer does what it was defined to do.
+- Removing the ability to run under a particular platform.
+
+### Examples of MINOR Command-Line Changes
+
+- Addition of a new option
+- Modifying the behavior of an option so it does the same thing better or more effectively or has added sub-options.
+- Adding support for running under an new platform
+
+### Examples of PATCH Command-Line changes
+
+- Fixing a bug in the behavior of an option.
+- Restoring an option to its original defined behavior.
+- Correcting spelling errors.
+
+## Engine APIs
+
+The **NUnit Engine** is intended to be used by third-party runners as well as our own **Console Runner**. Usage is through the defined API, specified by interfaces included in the `nunit.engine.api` assembly. It's important to note that public types and methods included in other assemblies are not considered a part of the published API. Direct creation and use of `nunit.engine`, `nunit.engine.core` or any other assemblies included in our packages is discouraged for that reason. See below for more about this.
+
+### Examples of MAJOR Api Changes
+
+- Removing a published interface.
+- Modifying an interface so that it breaks existing usage.
```

Comment:
It may be worth linking https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/breaking-change-rules.md for our own ease of reference to define what we mean by "breaks existing usage."